GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/SymbolicLinkTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.IO;
+using GVFS.FunctionalTests.FileSystemRunners;
+using GVFS.FunctionalTests.Should;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+
+namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
+{
+    // MacOnly until issue #297 (add SymLink support for Windows) is complete
+    [Category(Categories.MacOnly)]
+    [TestFixture]
+    public class SymbolicLinkTests : TestsWithEnlistmentPerFixture

```

Comment:
Do we need any tests the have symlinks to directories or directories to symlinks?  Also do we need any tests for syslinks and modified paths so files that are written to?

Reply:
> Do we need any tests the have symlinks to directories or directories to symlinks?

I think adding symlinks to directories is a good idea.  What did you have in mind for "directories to symlinks"?

> Also do we need any tests for syslinks and modified paths so files that are written to?

That's a good idea as well. Symlinks are added to ModifiedPaths.dat as soon as they're written by VFS for Git, and so I should be able to modify these tests to confirm that paths are being added there.

Reply:
Sorry what I meant on this was during a checkout when an item in the index can change from one type to another, so for example you are on a branch where `foo` is a directory and checkout another branch and now `foo` is a symlink.

Reply:
Ahh that makes sense thanks! I will add those tests as well.

Reply:
@kewillford I've added the tests and validate you suggested, and fixed a bug in `ModifiedPathsShouldContain` where `ModifiedPathsShouldContain` would succeed if a line existed that ended with the path being validated (e.g. ModifiedPathsShouldContain("foo") would pass if the file had the line `bar/foo`)

Reply:
Looks good thanks!